### ISSUE TITLE
Add support to pycryptodome

### ISSUE THREAD

**Issue by winstonma** [jpadilla/pyjwt#443]
I am using pycryptodome with pyjwt.

## Expected Result

Everything should work (if pycryptodome is supported)

## Actual Result

I got the following error

```
File "/Users/winstonma/Downloads/Ascope/app/venv/lib/python3.7/site-packages/jwt/contrib/algorithms/pycrypto.py", line 29, in prepare_key
    if isinstance(key, RSA._RSAobj):

AttributeError: module 'Crypto.PublicKey.RSA' has no attribute '_RSAobj'`
```

## Reproduction Steps

```python
import jwt
from jwt.contrib.algorithms.pycrypto import RSAAlgorithm

jwt.unregister_algorithm('RS256')
jwt.register_algorithm('RS256', RSAAlgorithm(RSAAlgorithm.SHA256))

...

payload = jwt.decode(token, key=secret_key, algorithms=['RS256'])
```

## System Information

    $ python -m jwt.help

```
{
  "cryptography": {
    "version": ""
  },
  "implementation": {
    "name": "CPython",
    "version": "3.7.4"
  },
  "platform": {
    "release": "18.7.0",
    "system": "Darwin"
  },
  "pyjwt": {
    "version": "1.7.1"
  }
}
```

This command is only available on PyJWT v1.6.3 and greater. Otherwise,
please provide some basic information about your system.

## Comment
After viewing the source code of pycryptodome, the root cause is the object _RSAobj is being repalced by RsaKey ([reference](https://github.com/Legrandin/pycryptodome/commit/8e4d6f3518697c9be0ec0078fed5163b4546b467#diff-e63e1e8d5b5365b9b84f14f4e9847c8c))

Is it possible that the pyjwt support both pycrypto and pycryptodome by adding support of RsaKey in pycrypto.py?

**Comment by jpadilla** [comment]
@winstonma what's the advantage of using pycryptodome over [cryptography](https://cryptography.io/en/latest/)?

**Comment by winstonma** [comment]
@jpadilla The main reason why I install [pycryptodome](https://pycryptodome.readthedocs.io) is because I got the following error message

```
  File "/dist-packages/jwt/api_jws.py", line 50, in register_algorithm
    raise ValueError('Algorithm already has a handler.')
ValueError: Algorithm already has a handler.
```

I installed [cryptography](https://cryptography.io/en/latest/) and it didn't fix my problem. I then installed the Pycrypto package and fixes my problem. So I assumed that I need to install Pycrypto in order to use pyjwt. Based on the assumption (Pycrypto is needed), I created this request as Pycrypto is deprecated somehow.
